frost-ristretto255/src/frost/keys.rs

```diff
@@ -0,0 +1,344 @@
+#[derive(Clone, Copy, Debug, Default, PartialEq)]
+pub struct Secret(pub(crate) Scalar);
+
+impl Secret {
+    /// Generates a new uniformly random secret value using the provided RNG.
+    pub fn random<R>(rng: &mut R) -> Self
+    where
+        R: CryptoRng + RngCore,
+    {
+        Self(Scalar::random(rng))
+    }
+}
+
+// Zeroizes `Secret` to be the `Default` value on drop (when it goes out of scope).  Luckily the
+// derived `Default` includes the `Default` impl of Scalar, which is four 0u64's under the hood.
+impl DefaultIsZeroes for Secret {}
+
+impl From<Scalar> for Secret {
+    fn from(source: Scalar) -> Secret {
+        Secret(source)
+    }
+}
+
+impl FromHex for Secret {
+    type Error = &'static str;
+
+    fn from_hex<T: AsRef<[u8]>>(hex: T) -> Result<Self, Self::Error> {
+        let mut bytes = [0u8; 32];
+
+        match hex::decode_to_slice(hex, &mut bytes[..]) {
+            Ok(()) => Secret::try_from(bytes),
+            Err(_) => Err("invalid hex"),
+        }
+    }
+}
+
+impl TryFrom<[u8; 32]> for Secret {
+    type Error = &'static str;
+
+    fn try_from(source: [u8; 32]) -> Result<Self, &'static str> {
+        match Scalar::from_canonical_bytes(source) {
+            Some(scalar) => Ok(Secret(scalar)),
+            None => Err("scalar was not canonically encoded"),
+        }
+    }
+}
+
+/// A public group element that represents a single signer's public key.
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct Public(pub(crate) RistrettoPoint);
+
+impl From<RistrettoPoint> for Public {
+    fn from(source: RistrettoPoint) -> Public {
+        Public(source)
+    }
+}
+
+impl From<Secret> for Public {
+    fn from(secret: Secret) -> Public {
+        Public(RISTRETTO_BASEPOINT_POINT * secret.0)
+    }
+}
+
+/// A Ristretto point that is a commitment to one coefficient of our secret
+/// polynomial.
+///
+/// This is a (public) commitment to one coefficient of a secret polynomial used
+/// for performing verifiable secret sharing for a Shamir secret share.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub(crate) struct CoefficientCommitment(pub(crate) RistrettoPoint);
+
+/// Contains the commitments to the coefficients for our secret polynomial _f_,
+/// used to generate participants' key shares.
+///
+/// [`ShareCommitment`] contains a set of commitments to the coefficients (which
+/// themselves are scalars) for a secret polynomial f, where f is used to
+/// generate each ith participant's key share f(i). Participants use this set of
+/// commitments to perform verifiable secret sharing.
+///
+/// Note that participants MUST be assured that they have the *same*
+/// [`ShareCommitment`], either by performing pairwise comparison, or by using
+/// some agreed-upon public location for publication, where each participant can
+/// ensure that they received the correct (and same) value.
+#[derive(Clone)]
+pub struct ShareCommitment(pub(crate) Vec<CoefficientCommitment>);
```

Comment:
A better name for this might be a VSSCommitment- it isn't really committing to a share, it is committing to the polynomial for which that share is a point on. So if the share is `f(i)`, the VSSCommitment is a vector of the coefficients of `f` "in the exponent", so that the verification step is just evaluating `f(i)` (again in the exponent) to make sure it matches the received share.

Reply:
Tried a similar rename: 046d797